core/src/main/java/org/apache/accumulo/core/data/impl/TabletIdImpl.java

```diff
@@ -36,7 +36,7 @@ public int compareTo(TabletId o) {
 
   @Override
   public Text getTableId() {
-    return new Text(ke.getTableId());
+    return new Text(ke.getTableId().getUtf8Bytes());
```

Comment:
Could also rely on the string constructor for Text here. It should use UTF8 bytes internally.